launcher/InstanceImportTask.cpp

```diff
@@ -439,25 +479,177 @@ void InstanceImportTask::processMultiMC()
+                QString hash;
+                QCryptographicHash::Algorithm hashAlgorithm;
+                hash = Json::ensureString(hashes, "sha1");
+                hashAlgorithm = QCryptographicHash::Sha1;
+                if (hash.isEmpty()) {
+                    hash = Json::ensureString(hashes, "sha512");
+                    hashAlgorithm = QCryptographicHash::Sha512;
+                    if (hash.isEmpty()) {
+                        hash = Json::ensureString(hashes, "sha256");
+                        hashAlgorithm = QCryptographicHash::Sha256;
+                        if (hash.isEmpty()) {
+                            throw JSONValidationError("No hash found for: " + file.path);
+                        }
+                    }
+                }
+                file.hash = QByteArray::fromHex(hash.toLatin1());
+                file.hashAlgorithm = hashAlgorithm;
+                // Do not use requireUrl, which uses StrictMode, instead use QUrl's default TolerantMode (as Modrinth seems to incorrectly
+                // handle spaces)
+                file.download = Json::requireString(Json::ensureArray(obj, "downloads").first(), "Download URL for " + file.path);
+                if (!file.download.isValid() || !Modrinth::validadeDownloadUrl(file.download)) {
+                    throw JSONValidationError("Download URL for " + file.path + " is not a correctly formatted URL");
+                }
+                files.push_back(file);
+            }
+
+            auto dependencies = Json::requireObject(obj, "dependencies", "modrinth.index.json");
+            for (auto it = dependencies.begin(), end = dependencies.end(); it != end; ++it) {
+                QString name = it.key();
+                if (name == "minecraft") {
+                    if (!minecraftVersion.isEmpty())
+                        throw JSONValidationError("Duplicate Minecraft version");
+                    minecraftVersion = Json::requireString(*it, "Minecraft version");
+                } else if (name == "fabric-loader") {
+                    if (!fabricVersion.isEmpty())
+                        throw JSONValidationError("Duplicate Fabric Loader version");
+                    fabricVersion = Json::requireString(*it, "Fabric Loader version");
+                } else if (name == "quilt-loader") {
+                    if (!quiltVersion.isEmpty())
+                        throw JSONValidationError("Duplicate Quilt Loader version");
+                    quiltVersion = Json::requireString(*it, "Quilt Loader version");
+                } else if (name == "forge") {
+                    if (!forgeVersion.isEmpty())
+                        throw JSONValidationError("Duplicate Forge version");
+                    forgeVersion = Json::requireString(*it, "Forge version");
+                } else {
+                    throw JSONValidationError("Unknown dependency type: " + name);
+                }
```

Comment:
nitpicky but, checking for duplicates is probably not needed, as json already disallows duplicate keys (thus it would have caused an error earlier)